bus-mapping/src/exec_trace.rs

```diff
@@ -14,6 +14,7 @@ impl fmt::Debug for OperationRef {
             "OperationRef{{ {}, {} }}",
             match self.0 {
                 Target::Start => "Start",
```

Comment:
The `Start` tag was already used as padding.  Is `Start` still necessary in the chunking approach?  I think we could remove `Start`.

From the PR description I see you're already aware of the missing checks to make sure that the chronological rw table doesn't contain extra entries not found in the evm circuit.  I think that can be solved with a very similar solution that we did with the Start tag.
Also I want to mention that I think it doesn't matter weather the padding rows are at the beginning or at the end (as long as they are forced to be consecutive).

Reply:
Agree! Right now It seems to me that we can remove `Start` and only use `Padding`.

Regarding to padding at beginning or end, I prefer `end` for the reason of `better coding style` in chronologically 
 rw_table handling. More elaboration, in later multiple chunk modification, I imagine the rw_table chunk should be pre-process in bus-mapping to chunk into BlockChunk structure
> while BlockChunk is almost the same as current `Block`, so the circuit assigment logic require less change

To literally represent `chronologically_rw_table` meaning and still make padding consecutive appearing, `chronologically_rw_table` should be sorted by `rw_counter` in ascending order **inside** chunk , while other non-padding rw rw_counter still start from uint`1`.  Therefore padding step rw_counter better assign largerly to make it consecutive appear at the end.

Reply:
@ed255 based on my reasoning above, padding at the end make life easier.
Then I realized we need add `Padding` (largest enum), and also need to keep `Start` (smallest enum) and make it appear in first chunk first row. Because first chunk first row is public input and we will design it like constant commitment in root circuit, so we need smallest enum `Start` to assure it can appear in the first row of `address-sorted-rwtable`

Reply:
I see, so there's 1 single `Start` row at the top, and a variable number of `Padding` rows at the bottom, right?

Reply:
> I see, so there's 1 single `Start` row at the top, and a variable number of `Padding` rows at the bottom, right?

Exactly!